src/main/java/seedu/address/logic/commands/AddAppCommand.java

```diff
@@ -28,31 +33,73 @@
             + PREFIX_START + "01/11/19 1800 "
             + PREFIX_END + "01/11/19 1900";
 
-    public static final String MESSAGE_SUCCESS = "New appointment added: %1$s";
+    public static final String MESSAGE_USAGE_RECURSIVELY = COMMAND_WORD + ": Adds recursively appointment"
+            + " to the address book. \n"
+            + "Parameters: "
+            + PREFIX_ID + "REFERENCE ID "
+            + "[" + PREFIX_RECURSIVE + "PREFIX_RECURSIVE w/m/y] "
+            + "[" + PREFIX_RECURSIVE_TIMES + "PREFIX_RECURSIVE_TIMES] "
+            + PREFIX_START + "PREFIX_EVENT "
+            + PREFIX_END + "PREFIX_EVENT \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ID + "001A "
+            + PREFIX_RECURSIVE + "m "
+            + PREFIX_RECURSIVE_TIMES + "2 "
+            + PREFIX_START + "01/11/19 1800 "
+            + PREFIX_END + "01/11/19 1900";
+
+    public static final String MESSAGE_SUCCESS = "Appointment added: %1$s";
+    public static final String MESSAGE_SUCCESS_RECURSIVE = " recusive Appointments were added";
     public static final String MESSAGE_DUPLICATE_EVENT = "This appointment is already scheduled.";
     public static final String MESSAGE_CLASH_APPOINTMENT = "This appointment clashes with a pre-existing appointment.";
 
     private final Event toAdd;
+    private final List<Event> eventList;
 
     /**
      * Creates an AddAppCommand to add the specified {@code Person}
      */
     public AddAppCommand(Event toAdd) {
         requireNonNull(toAdd);
         this.toAdd = toAdd;
+        this.eventList = null;
+
+    }
+
+    public AddAppCommand(List<Event> eventList) {
+        requireNonNull(eventList);
+        this.toAdd = null;
+        this.eventList = eventList;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        if (eventList == null) {
+            addOneEvent(model);
+            model.updateFilteredEventList(toAdd.getPersonId());
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+
+        } else {
+            for (Event e : eventList) {
+                if (model.hasEvent(e)) {
+                    throw new CommandException(MESSAGE_DUPLICATE_EVENT);
```

Comment:
Could you clarify in the UG and DG, what would happen if only 1 of mutliple recurring events conflict with another event? Should it all fail? Or should only that single recurring event fail?